.chloggen/fix-gzip-empty-bug.yaml

```diff
@@ -0,0 +1,19 @@
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: splunkhecexporter
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: While sending empty data to Splunk with compression enabled, the gzip buffer will be at least 18 bytes (10-byte header + 0-byte payload + 8-byte footer), even if uncompressed data is of 0 bytes.
+HEC will return with '{\\\"text\\\":\\\"No data\\\",\\\"code\\\":5}'' if uncompressed data of 0 bytes is gzipped before sending to HEC. 
+The current check in the overall buffer, regardless of compression. So we need to add another check if compression is enabled.
```

Comment:
I think you need to use a different approach for this note. Please try to keep it to one line and move the content to the subtext. Use the `|` symbol there to have multiple lines.